### ISSUE TITLE
Allow Modules to Declare Layout Templates as Dependencies

### ISSUE THREAD

**Issue by stpaultim** [backdrop/backdrop-issues#6593]
## Description of the need

I'm creating a recipe module and would like my recipe to require a specific layout template and automatically enable that layout template as the default layout template. 

Apparently, modules can only list other modules as dependencies. If I am wrong about this, let me know. 

## Proposed solution

Allow modules to declare layout templates as dependencies

## Alternatives that have been considered

??

## Additional information

There is a similar feature request for themes.
https://github.com/backdrop/backdrop-issues/issues/5316 Allow themes to declare dependencies on modules and layout templates

## Draft of feature description for Press Release (1 paragraph at most)

Backdrop now includes the ability for a module to declare a dependency on a specific layout template.

**Comment by klonos** [comment]
Yeah, I think that any type of project should be able to declare dependency on any other type of project. So that:
- modules can depend on themes and/or layout templates
- themes can depend on modules and/or layout templates
- layout templates can depend on modules and/or themes

If the code where each of that is different, then it does make sense to have different issues for these. But if the code is the same, as I am suspecting (the installer module basically), then we might wanna consider consolidating all these into a single request.

**Comment by olafgrabienski** [comment]
> I'm creating a recipe module and would like my recipe to
> - require a specific layout template
> - automatically enable that layout template
> -as the default layout template

(I've broken up the first sentence to emphasize how I understand it.)

While I agree with the first and the second feature, I'm skeptical about the third. It might work for new sites but not for existing ones. Existing sites are already using the default layout, and changing this to another template could change its appearance pretty much.

**Comment by klonos** [comment]
Yup, downloading and enabling layout templates should be relatively easy to achieve (actually, templates are always "enabled" - but you can hide them from the setting page in `admin/structure/layouts/settings`). Creating a new layout and using a template for it should also be easy and harmless. However setting the template as the default, or replacing the template of any existing layout should be done with caution. There can be a confirmation step in the installer that could provide the option to do the replacement, but there should be a warning to state that things are likely to break.

**Comment by stpaultim** [comment]
Running into this again. Creating a feature in a module that requires a layout that can expand the content region to 100% of the available screen. The only choice I can see, other than encouraging the use of a contrib theme is to include a layout that overrides an existing layout. This seems like a pretty heavy approach and one with implications if the site is already using that template. 

I suppose I could include a template override that ONLY expands if a particular class is present.
